packages/api-server/src/methods/modules/eth.ts

```diff
@@ -1114,8 +1114,34 @@ export class Eth {
         return undefined;
     }
 
+    // handle block hash
+    if (
+      typeof blockParameter === "object" &&
+      blockParameter.blockHash != null
+    ) {
+      const block = await this.getBlockByHash([
+        blockParameter.blockHash,
+        false,
+      ]);
+      if (block == null) {
+        throw new InvalidParamsError(
```

Comment:
Returns HeaderNotFound?

Reply:
I think `HeaderNotFound == tipBlockNumber < user request's block number` ?

Reply:
> I think `HeaderNotFound == tipBlockNumber < user request's block number` ?

This meaning cannot be derived from "HeaderNotFound".

"xxxNotFound" indicates that we cannot find the corresponding resource from database.

Reply:
updated